xarray/core/merge.py

```diff
@@ -1,173 +1,368 @@
-from .alignment import align, partial_align, align_variables
-from .utils import ChainMap
-from .variable import as_variable
+from collections import defaultdict, deque
+
+import pandas as pd
+
+from .alignment import align, deep_align
+from .utils import ChainMap, Frozen
+from .variable import as_variable, Coordinate, Variable, default_index_coordinate
```

Comment:
The following are imported but are not used: `defaultdict, deque, align, ChainMap, Coordinate, Variable iteritems`